ddtrace/tracer/option.go

```diff
@@ -186,16 +187,22 @@ func WithPropagator(p Propagator) StartOption {
 	}
 }
 
-// WithServiceName sets the default service name to be used with the tracer. It is
-// deprecated in favour of WithService and will be removed in the next major version.
+// WithServiceName sets the default service name to be used with the tracer. This will
+// not affect the service name used by any integrations. It is deprecated in favour of
+// WithService and will be removed in the next major version.
 func WithServiceName(name string) StartOption {
-	return WithService(name)
+	return func(c *config) {
+		c.serviceName = name
+	}
 }
 
-// WithService sets the default service name to be used with the tracer.
+// WithService sets the default service name to be used with the tracer. It will also be
+// used as the service name for server and framework integrations
+// (e.g. http servers, grpc, etc.)
 func WithService(name string) StartOption {
 	return func(c *config) {
 		c.serviceName = name
+		globalconfig.SetServiceName(c.serviceName)
```

Comment:
:+1: Okay makes sense. Basically the new functionality for "core service" comes from checking/using `globalconfig.ServiceName()`.